Updated about RHDH (#1157)
redhat-developer/red-hat-developers-documentation-rhdh: modules/about/con_understanding-internal-developer-platforms.adoc

:_newdoc-version: 2.18.3
:_template-generated: 2025-05-12

:_mod-docs-content-type: CONCEPT

[id="understanding-internal-developer-platforms_{context}"]
= Understanding internal developer platforms

An internal developer platform (IDP) is a curated set of tools and services that supports developer self-service. Instead of navigating multiple systems, developers use a unified interface to provision environments, deploy code, and access APIs.

Why IDPs matter::

IDPs address the challenges of modern software delivery by enabling self-service, enforcing standards, and improving the developer experience.

For organizations::

* *Scalability:* {product-very-short} enables consistent developer onboarding and application delivery across growing teams and environments.
* *Security:* Role-based access control (RBAC) and integration with enterprise systems ensure access is managed securely and in line with compliance requirements.
* *Operational efficiency:* By removing manual handoffs and centralizing key development workflows, {product-very-short} improves time to value and increases return on engineering investment.

For platform engineers::

> * *Curated platforms:* Platform teams can design reusable templates and integrations aligned with organizational policies and developer needs.

Reviewer: Nice user-focused bullet here

Plugins and integrations::
Extend {product-very-short} with verified plugins that add new functionality without downtime. Dynamically integrate with supported tools such as Tekton for pipelines, GitOps for deployment automation, Nexus Repository for artifact storage, and JFrog Artifactory. {product-very-short} also supports connecting to {ocp-brand-name}, CI/CD systems, and security scanners through Red Hat-curated extensions.

Role-Based Access Control (RBAC)::
Manage user access with robust security permissions tailored to organizational needs.

Software catalog::
Search, view, and manage services, APIs, and libraries from a central inventory. Track ownership, metadata, and component health in one place.

Software templates::
Accelerate project setup using preconfigured templates for CI/CD, runtime, and security. Standardize implementation while enabling developer autonomy.

Tech docs::
Create, store, and view technical documentation alongside code. Make content searchable, consistently formatted, and accessible through the portal.

Scalability::

> Support growing teams and applications while maintaining access to the same tools and services.

Reviewer: Should we make this a reference module?

Author: I think this is more of a concept. We are informing users about Key features.

Reviewer: Sure, I think you could make a case either way. Here is what Gemini suggested: 

> A list of key features is best suited to be part of a reference module, rather than a concept, when used in the context of a software project or product documentation. Reference modules serve as a repository of information that can be reused across different contexts or applications, making them ideal for documenting key features. 
> 
> Concepts are more abstract and general ideas that can be used to categorize or organize information. While they can be used in conjunction with reference modules, a list of key features is a specific piece of information that is best handled as a reusable component within a reference module.

Author: Thank you!
Made it a block heading.

[role="_additional-resources"]
.Additional resources

* link:https://docs.redhat.com/documentation/red_hat_developer_hub/{product-version}/html/customizing_red_hat_developer_hub/configuring-templates[Configuring templates]

> * link:https://docs.redhat.com/documentation/red_hat_developer_hub/{product-version}/html/techdocs_for_red_hat_developer_hub/index[TechDocs for Red Hat Developer Hub]

Reviewer: Replace product names with attributes

:_newdoc-version: 2.18.3
:_template-generated: 2025-05-12

:_mod-docs-content-type: CONCEPT

[id="understanding-internal-developer-platforms_{context}"]
= Understanding internal developer platforms

An internal developer platform (IDP) is a curated set of tools and services that supports developer self-service. Instead of navigating multiple systems, developers use a unified interface to provision environments, deploy code, and access APIs.

Why IDPs matter::

IDPs address the challenges of modern software delivery by enabling self-service, enforcing standards, and improving the developer experience.

For developers::

> * *Faster onboarding:* Developers can use learning paths, software templates, and the software catalog to deploy compliant services within minutes—without depending on other teams for setup.

Reviewer 2: ```suggestion
* *Faster onboarding:* Developers can use learning paths, software templates, and software catalog to deploy compliant services within minutes, without depending on other teams for setup.
```

* *Reduced cognitive load:* Tools, documentation, and deployment environments are available in one place, eliminating the need to switch between systems or manage disconnected resources.
* *Self-service workflows:* Developers can create applications or environments on demand, without raising tickets or waiting for approvals.
* *Built-in standards:* Platform-defined templates and software templates ensure secure and compliant workflows by default.

> * *Cross-team visibility:* Shared service catalogs and documentation improve discoverability and reuse across teams.

Reviewer 2: ```suggestion
* *Cross-team visibility:* Shared service catalogs and documentation improve discoverability and promote reuse across teams.
```

* *Higher productivity:* Developers spend more time building features and less time configuring infrastructure or resolving toolchain inconsistencies.

> For platform engineers::

Reviewer 2: I was thinking the organizations section might come first followed by platform engineers and developers section. wdyt?

* *Curated platforms:* Platform teams can design reusable templates and integrations aligned with organizational policies and developer needs.
* *Central configuration:* Infrastructure and policies are defined as code and centrally managed, reducing drift and maintenance overhead.
* *Governance at scale:* Policies and best practices are embedded into developer workflows using automation and templates, without adding friction to the process.

For organizations::

* *Scalability:* {product-very-short} enables consistent developer onboarding and application delivery across growing teams and environments.
* *Security:* Role-based access control (RBAC) and integration with enterprise systems ensure access is managed securely and in line with compliance requirements.
* *Operational efficiency:* By removing manual handoffs and centralizing key development workflows, {product-very-short} improves time to value and increases return on engineering investment.

== Key features

Centralized dashboard::

> Access development tools, CI/CD pipelines, APIs, monitoring tools, and documentation from a single interface. Integrate with systems like Git, {ocp-brandname}, Kubernetes, and JIRA.

Reviewer 2: ```suggestion
Access development tools, CI/CD pipelines, APIs, monitoring tools, and documentation from a single interface. Integrate with systems like Git, {ocp-brand-name}, Kubernetes, and JIRA.
```

Configuration management::
Manage updates and configurations through centralized repositories, improving version control and consistency across environments.

Learning paths::

> Guide developers through structured tutorials and onboarding steps. Help teams upskill with internal and Red Hat training resources in one place. For more information, see []

Reviewer 2: Reference link is missing here.

Plugins and integrations::

> Extend {product-very-short} with verified plugins that add new functionality without downtime. Dynamically integrate with supported tools such as Tekton for pipelines, GitOps for deployment automation, Nexus Repository for artifact storage, and JFrog Artifactory. {product-very-short} also supports connecting to {ocp-brandname}, CI/CD systems, and security scanner through Red Hat-curated extensions.

Reviewer 2: ```suggestion
Extend {product-very-short} with verified plugins that add new functionality without downtime. Dynamically integrate with supported tools such as Tekton for pipelines, GitOps for deployment automation, Nexus Repository for artifact storage, and JFrog Artifactory. {product-very-short} also supports connecting to {ocp-brand-name}, CI/CD systems, and security scanners through Red Hat-curated extensions.
```

Role-Based Access Control (RBAC)::
Manage user access with robust security permissions tailored to organizational needs.

Software catalog::
Search, view, and manage services, APIs, and libraries from a central inventory. Track ownership, metadata, and component health in one place.

Software templates::
Accelerate project setup using preconfigured templates for CI/CD, runtime, and security. Standardize implementation while enabling developer autonomy.

Tech docs::
Create, store, and view technical documentation alongside code. Make content searchable, consistently formatted, and accessible through the portal.

Scalability::
Support growing teams and applications while maintaining access to the same tools and services.

[role="_additional-resources"]
.Additional resources

> * For more information about software templates, see link:https://docs.redhat.com/documentation/red_hat_developer_hub/{product-version}/html/customizing_red_hat_developer_hub/configuring-templates[Configuring templates].

Reviewer 2: Consider removing the leading sentence in the additional resources section. Perhaps we could just add the text embedded with links, similar to the format most CCS follow.
```suggestion
* link:https://docs.redhat.com/documentation/red_hat_developer_hub/{product-version}/html/customizing_red_hat_developer_hub/configuring-templates[Configuring templates].
```

For developers::

* *Faster onboarding:* Developers can use learning paths, software templates, and software catalog to deploy compliant services within minutes, without depending on other teams for setup.
* *Reduced cognitive load:* Tools, documentation, and deployment environments are available in one place, eliminating the need to switch between systems or manage disconnected resources.
* *Self-service workflows:* Developers can create applications or environments on demand, without raising tickets or waiting for approvals.
* *Built-in standards:* Platform-defined templates and software templates ensure secure and compliant workflows by default.
* *Cross-team visibility:* Shared service catalogs and documentation improve discoverability and promote reuse across teams.
* *Higher productivity:* Developers spend more time building features and less time configuring infrastructure or resolving toolchain inconsistencies.

== Key features

Centralized dashboard::
Access development tools, CI/CD pipelines, APIs, monitoring tools, and documentation from a single interface. Integrate with systems like Git, {ocp-brand-name}, Kubernetes, and JIRA.

> Configuration management::

Reviewer 3: This isn't an RHDH feature IMO.
